src/utils/symbol_utils.ml

```diff
@@ -8,18 +8,76 @@ type symbol =
   }
 
 
-let find_symbol program name =
+type extern_symbol =
+  {
+    tid : tid;
+    address : string;
+    name : string;
+    cconv : string option;
+    args : (Var.t * Exp.t * intent option) list;
+  }
+
+
+let extern_symbols = ref []
+
+
+let get_project_calling_convention (project : Project.t) : string option =
+  Project.get project Bap_abi.name
+
+
+let build_extern_symbols (project : Project.t) (program : program term) : unit =
+  let parsed_symbols = String.Set.to_list (Cconv.parse_dyn_syms project) in
+  let calling_convention = get_project_calling_convention project in
+  let tid_map = Address_translation.generate_tid_map program in
```

Comment:
Right now generating the TID map is computationally expensive. It should be a parameter to the function, so that we only have to compute the map once.